bftclient/src/quorums.cpp

```diff
@@ -0,0 +1,92 @@
+
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License"). You may not use
+// this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license
+// terms. Your use of these subcomponents is subject to the terms and conditions of the
+// subcomponent's license, as noted in the LICENSE file.
+
+#include <algorithm>
+
+#include "bftclient/quorums.h"
+#include "bftclient/exception.h"
+
+namespace bft::client {
+
+MofN QuorumConverter::ToMofN(const LinearizableQuorum& quorum) const {
+  MofN new_quorum;
+  new_quorum.wait_for = 2 * f_val_ + c_val_ + 1;
+  if (quorum.destinations.empty()) {
+    // If the user doesn't provide destinations, send to all replicas
+    new_quorum.destinations = all_replicas_;
+  } else if (quorum.destinations.size() < new_quorum.wait_for) {
+    throw BadQuorumConfigException(
+        "Destination does not contain enough replicas for a linearizable quorum. Destination size: " +
+        std::to_string(quorum.destinations.size()) +
+        "is less than 2f + c + 1 = " + std::to_string(new_quorum.wait_for));
```

Comment:
This could be defined once as a constant string to simplify a change if required.

Reply:
I't be pretty strange for this to change, unless we changed protocols away from SBFT. Changing it inline doesn't feel like a burden.